src/backend/distributed/commands/statistics.c

```diff
@@ -445,19 +456,32 @@ GetExplicitStatisticsCommandList(Oid relationId)
 
 		if (alterStatisticsOwnerCommand != NULL)
 		{
-			alterStatisticsCommandList =
-				lappend(alterStatisticsCommandList,
+			explicitStatisticsCommandList =
+				lappend(explicitStatisticsCommandList,
 						makeTableDDLCommandString(alterStatisticsOwnerCommand));
 		}
 	}
 
+	/* find indexes on current relation, in case of modified stats target */
+	Relation relation = relation_open(relationId, AccessShareLock);
+	List *indexOidList = RelationGetIndexList(relation);
+
+	Oid indexId = InvalidOid;
+	foreach_oid(indexId, indexOidList)
+	{
+		/* we need alter index commands for altered targets on expression indexes */
+		List *alterIndexStatisticsCommands = GetAlterIndexStatisticsCommands(indexId);
+
+		explicitStatisticsCommandList =
+			list_concat(explicitStatisticsCommandList, alterIndexStatisticsCommands);
+	}
+
+	relation_close(relation, AccessShareLock);
```

Comment:
https://github.com/citusdata/citus/pull/4455#discussion_r549608956

moving `relation_close(relation, AccessShareLock);` to somewhere else wouldn't work, we should actually use `NoLock` here.

Also, previous place was better.